Neos.EventSourcedNeosAdjustments/Classes/EventSourcedRouting/Projection/DocumentUriPathFinder.php

```diff
@@ -0,0 +1,146 @@
+{
+    private Connection $dbal;
+
+    private ?ContentStreamIdentifier $liveContentStreamIdentifierRuntimeCache = null;
+
+    public function __construct(Connection $dbal)
+    {
+        $this->dbal = $dbal;
+    }
+
+    /**
+     * @param NodeName $siteNodeName
+     * @param string $uriPath
+     * @param string $dimensionSpacePointHash
+     * @return DocumentNodeInfo
+     * @throws NodeNotFoundException
+     */
+    public function getEnabledBySiteNodeNameUriPathAndDimensionSpacePointHash(NodeName $siteNodeName, string $uriPath, string $dimensionSpacePointHash): DocumentNodeInfo
+    {
+        return $this->fetchSingle('dimensionSpacePointHash = :dimensionSpacePointHash AND siteNodeName = :siteNodeName AND uriPath = :uriPath AND disabled = 0', compact('dimensionSpacePointHash', 'siteNodeName', 'uriPath'));
+    }
+
+    /**
+     * @param NodeAggregateIdentifier $nodeAggregateIdentifier
+     * @param string $dimensionSpacePointHash
+     * @return DocumentNodeInfo
+     * @throws NodeNotFoundException
+     */
+    public function getOneByIdAndDimensionSpacePointHash(NodeAggregateIdentifier $nodeAggregateIdentifier, string $dimensionSpacePointHash): DocumentNodeInfo
+    {
+        return $this->fetchSingle('nodeAggregateIdentifier = :nodeAggregateIdentifier AND dimensionSpacePointHash = :dimensionSpacePointHash', compact('nodeAggregateIdentifier', 'dimensionSpacePointHash'));
+    }
+
+    /**
+     * @param DocumentNodeInfo $nodeInfo
+     * @return DocumentNodeInfo
+     * @throws NodeNotFoundException
+     */
+    public function getParentNode(DocumentNodeInfo $nodeInfo): DocumentNodeInfo
+    {
+        return $this->getOneByIdAndDimensionSpacePointHash($nodeInfo->getParentNodeAggregateIdentifier(), $nodeInfo->getDimensionSpacePointHash());
+    }
+
+    /**
+     * @param NodeAggregateIdentifier $succeedingNodeAggregateIdentifier
+     * @param NodeAggregateIdentifier $parentNodeAggregateIdentifier
+     * @param string $dimensionSpacePointHash
+     * @return DocumentNodeInfo
+     * @throws NodeNotFoundException
+     */
+    public function getPrecedingNode(NodeAggregateIdentifier $succeedingNodeAggregateIdentifier, NodeAggregateIdentifier $parentNodeAggregateIdentifier, string $dimensionSpacePointHash): DocumentNodeInfo
+    {
+        return $this->fetchSingle('dimensionSpacePointHash = :dimensionSpacePointHash AND parentNodeAggregateIdentifier = :parentNodeAggregateIdentifier AND succeedingNodeAggregateIdentifier = :succeedingNodeAggregateIdentifier', compact('dimensionSpacePointHash', 'parentNodeAggregateIdentifier', 'succeedingNodeAggregateIdentifier'));
+    }
+
+    /**
+     * @param NodeAggregateIdentifier $parentNodeAggregateIdentifier
+     * @param string $dimensionSpacePointHash
+     * @return DocumentNodeInfo
+     * @throws NodeNotFoundException
+     */
+    public function getFirstEnabledChildNode(NodeAggregateIdentifier $parentNodeAggregateIdentifier, string $dimensionSpacePointHash): DocumentNodeInfo
```

Comment:
ah. that means the other methods are ignoring whether the node is enabled or not? I'd write this in comments on the other methods.

Reply:
True, I added doc comments to all public finder methods